### PR TITLE
Enable nullability in ToolStripDropDownItem

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in ToolStripDropDownItem.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8065)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripDropDownItem.cs [hunk 1]
```diff
@@ -56,55 +58,56 @@ protected ToolStripDropDownItem(string text, Image image, params ToolStripItem[]
...
             {
-                if (dropDown is null)
+                if (_dropDown is null)
                 {
                     DropDown = CreateDefaultDropDown();
-                    if (!(this is ToolStripOverflowButton))
+                    if (this is not ToolStripOverflowButton)
                     {
-                        dropDown.SetAutoGeneratedInternal(true);
+                        _dropDown!.SetAutoGeneratedInternal(true);
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripDropDownItem.cs:71]: Where is `_dropDown` set in this block? it entered when it is null.

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStripDropDownItem.cs:71]: Here: `DropDown = CreateDefaultDropDown();`

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStripDropDownItem.cs:71]: I wonder if we can apply `MemberNotNull(nameof(_dropDown))` to the setter... 💭

**Review by RussKie — COMMENTED** [review]
Looks good modulo `DropDown` API questions

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripDropDownItem.cs (hunk 1 above)
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStripDropDownItem.cs:71]: The problem is that DropDown can also be set to `null`. So, we cannot use `MemberNotNull(nameof(_dropDown))`. There is a pending TODO for CA2245 warnings in CodeAnalysis.src.globalconfig that if I am not mistaken should address such cases.

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStripDropDownItem.cs:71]: Ah, yes. Thank you for confirming it.

**Review by RussKie — APPROVED** [review]

**Pull request merged**